src/persistence.rs

```diff
@@ -6,92 +6,13 @@
-use crate::{
-    hd::KeyTree, loader::KeystoreLoader, txn_builder::TransactionState, EncryptingResourceAdapter,
-    KeystoreError, KeystoreState,
-};
-use arbitrary::{Arbitrary, Unstructured};
-use async_std::sync::Arc;
+use crate::{hd::KeyTree, loader::KeystoreLoader, EncryptingResourceAdapter, KeystoreError};
 use atomic_store::{load_store::BincodeLoadStore, AtomicStoreLoader, RollingLog};
-use derivative::Derivative;
-use espresso_macros::ser_test;
-use key_set::{OrderByOutputs, ProverKeySet};
 use reef::*;
-use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Serialize};
 use snafu::ResultExt;
 
-const ATOMIC_STORE_RETAINED_ENTRIES: u32 = 5;
-
-// Serialization intermediate for the static part of a KeystoreState.
-#[derive(Deserialize, Serialize, Debug)]
-struct KeystoreStaticState<'a> {
-    #[serde(with = "serde_ark_unchecked")]
-    proving_keys: Arc<ProverKeySet<'a, OrderByOutputs>>,
-}
-
-impl<'a, L: Ledger> From<&KeystoreState<'a, L>> for KeystoreStaticState<'a> {
-    fn from(w: &KeystoreState<'a, L>) -> Self {
-        Self {
-            proving_keys: w.proving_keys.clone(),
-        }
-    }
-}
-
-mod serde_ark_unchecked {
-    use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
-    use serde::{
-        de::{Deserialize, Deserializer},
-        ser::{Serialize, Serializer},
-    };
-    use std::sync::Arc;
-
-    pub fn serialize<S: Serializer, T: CanonicalSerialize>(
-        t: &Arc<T>,
-        s: S,
-    ) -> Result<S::Ok, S::Error> {
-        let mut bytes = Vec::new();
-        t.serialize_unchecked(&mut bytes).unwrap();
-        Serialize::serialize(&bytes, s)
-    }
-
-    pub fn deserialize<'a, D: Deserializer<'a>, T: CanonicalDeserialize>(
-        d: D,
-    ) -> Result<Arc<T>, D::Error> {
-        let bytes = <Vec<u8> as Deserialize<'a>>::deserialize(d)?;
-        Ok(Arc::new(T::deserialize_unchecked(&*bytes).unwrap()))
-    }
-}
-
-// Serialization intermediate for the dynamic part of a KeystoreState.
-#[ser_test(arbitrary, types(cap::Ledger), ark(false))]
-#[derive(Debug, Derivative, Deserialize, Serialize)]
-#[derivative(PartialEq(bound = "L: Ledger"))]
-#[serde(bound = "")]
-struct KeystoreSnapshot<L: Ledger> {
-    txn_state: TransactionState<L>,
-}
-
-impl<'a, L: Ledger> From<&KeystoreState<'a, L>> for KeystoreSnapshot<L> {
-    fn from(w: &KeystoreState<'a, L>) -> Self {
-        Self {
-            txn_state: w.txn_state.clone(),
-        }
-    }
-}
-
-impl<'a, L: Ledger> Arbitrary<'a> for KeystoreSnapshot<L>
-where
-    TransactionState<L>: Arbitrary<'a>,
-    TransactionHash<L>: Arbitrary<'a>,
-{
-    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
-        Ok(Self {
-            txn_state: u.arbitrary()?,
-        })
-    }
-}
-
-pub struct AtomicKeystoreStorage<'a, L: Ledger, Meta: Serialize + DeserializeOwned> {
+pub struct AtomicKeystoreStorage<Meta: Serialize + DeserializeOwned> {
```

Comment:
Since this is now only storing metadata, I think a rename is in order.

Reply:
Renamed it to `MetaStore`. Also renamed the file.